pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1071,13 +1062,15 @@ def _check_exception_inherit_from_stopiteration(
         return any(_class.qname() == stopiteration_qname for _class in exc.mro())
 
     def _check_consider_using_comprehension_constructor(self, node: nodes.Call) -> None:
-        if (
-            isinstance(node.func, nodes.Name)
-            and node.args
-            and isinstance(node.args[0], nodes.ListComp)
-        ):
-            if node.func.name == "dict":
-                element = node.args[0].elt
+        match node:
+            case nodes.Call(
+                func=nodes.Name(name=name), args=[nodes.ListComp(elt=element), *_]
+            ):
+                pass
+            case _:
+                return
+        match name:
+            case "dict":
```

Comment:
I reverted the `call_name` one but left this in place. With just `match name`, `case "dict": ...` and `case "set": ...` I think match is an improvement.